lotman/lotman_linux.go

```diff
@@ -314,9 +333,580 @@ func getFederationIssuer() (string, error) {
+		if strings.HasPrefix(nsAd.Path, "/pelican/monitoring") {
+			continue
+		}
+		var issuer string
+		if len(nsAd.Issuer) > 0 {
+			issuer = (nsAd.Issuer[0]).IssuerUrl.String()
+		} else {
+			issuer = federationIssuer
+		}
+
+		directorLotMap[nsAd.Path] = Lot{
+			LotName: nsAd.Path,
+			Owner:   issuer, // grab the first issuer -- lotman doesn't currently support multiple direct owners
+			// Assign parent as the root lot at the cache. This lets root edit the lot, but still allows the owner of the namespace to further subdivide
+			Parents: []string{"root"},
+			Paths: []LotPath{
+				{
+					Path: 	nsAd.Path,
+					Recursive: true,
+				},
+			},
+		}
+	}
+
+	return directorLotMap, nil
+}
+
+// One limitation in Lotman is that a lot cannot be created unless all of its parents exist. Unfortunately,
+// this means we have to sort our lots topologically to ensure that we create them in the correct order.
+// Failure to do so appears to result in a segfault in Lotman.
+func topoSort(lotMap map[string]Lot) ([]Lot, error) {
+    sorted := make([]Lot, 0, len(lotMap))
+    visited := make(map[string]bool)
+
+	// Recursively visit each lot and its parents, DFS-style
+	var visit func(string) error
+    visit = func(name string) error {
+        if visited[name] {
+            return nil
+        }
+        visited[name] = true
+
+        // Visit all parents first
+        for _, parent := range lotMap[name].Parents {
+            if err := visit(parent); err != nil {
+                return err
+            }
+        }
+		// Adding the leaves of the DFS parent tree to the sorted list
+		// guarantees that we'll add the parents before the children
+        sorted = append(sorted, lotMap[name])
+        return nil
+    }
+
+    for name := range lotMap {
+        if err := visit(name); err != nil {
+            return nil, err
+        }
+    }
+
+    return sorted, nil
+}
+
+
+// Initialize the lot configurations based on provided policy, discovered namespaces,
+// and available cache space, handling any necessary merges and validations along the way.
+func initLots(nsAds []server_structs.NamespaceAdV2) ([]Lot, error) {
+	var internalLots []Lot
+
+	policies, err := getPolicyMap()
+	if err != nil {
+		return internalLots, errors.Wrap(err, "unable to parse lotman configuration")
+	}
+
+	// Get the configured policy, which defines any lots we may need to handle
+	// along with merging logic and purge ordering
+	policyName := param.Lotman_EnabledPolicy.GetString()
+	if _, exists := policies[policyName]; !exists {
+		return internalLots, errors.Errorf("enabled policy %s is not defined in the configuration", policyName)
+	}
+	policy := policies[policyName]
+
+	discoverPrefixes := policy.DiscoverPrefixes
+	shouldMerge := policy.MergeLocalWithDiscovered
+	if shouldMerge && !discoverPrefixes {
+		return internalLots, errors.New("MergeLocalWithDiscovered is set to true, but DiscoverPrefixes is set to false. This is not a valid configuration")
```

Comment:
I view it as a more assertive check. True, if nothing is discovered, the merge is a no-op, but such configuration probably means the administrator doesn't really understand what these options do, or doesn't realize they have it set up this way.

I'm not _entirely_ opposed to making this a warning instead of an error, but my experience working with the folks who run our software in production is that none of them bother to look at startup logs unless there's an issue that prevents startup. I still see individuals using `Origin.NamespacePrefix` despite the fact that we deprecated nearly a year ago with a very easy drop-in replacement. 

If you still want to see the warning instead of a fatal error, I'll make the change, but I think this will wind up going unnoticed.